spark/src/test/scala/org/apache/spark/sql/delta/IdentityColumnIngestionSuite.scala

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import java.io.PrintWriter
+
+import org.apache.spark.sql.delta.GeneratedAsIdentityType.{GeneratedAlways, GeneratedByDefault}
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.execution.streaming.MemoryStream
+import org.apache.spark.sql.types._
+
+/**
+ * Identity Column test suite for ingestion, including insert-only MERGE.
+ */
+trait IdentityColumnIngestionSuiteBase extends IdentityColumnTestUtils {
+
+  import testImplicits._
+
+  private val tblName = "identity_test"
+  private val tempTblName = "identity_test_temp"
+  private val tempCsvFileName = "test.csv"
+
+  /** Helper function to write a single 'value' column into `sourcePath`. */
+  private def setupSimpleCsvFiles(sourcePath: String, start: Int, end: Int): Unit = {
+    val writer = new PrintWriter(s"$sourcePath/$tempCsvFileName")
+    // Write header.
+    writer.write("value\n")
+    // Write values.
+    (start to end).foreach { v =>
+      writer.write(s"$v\n")
+    }
+    writer.close()
+  }
+
+  object IngestMode extends Enumeration {
+    // Ingest using data frame append v1.
+    val appendV1 = Value
+
+    // Ingest using data frame append v2.
+    val appendV2 = Value
+
+    // Ingest using "INSERT INTO ... VALUES".
+    val insertIntoValues = Value
+
+    // Ingest using "INSERT INTO ... SELECT ...".
+    val insertIntoSelect = Value
+
+    // Ingest using "INSERT OVERWRITE ... VALUES".
+    val insertOverwriteValues = Value
+
+    // Ingest using "INSERT OVERWRITE ... SELECT ...".
+    val insertOverwriteSelect = Value
+
+
+    // Ingest using streaming query.
+    val streaming = Value
+
+    // Ingest using MERGE INTO ... WHEN NOT MATCHED INSERT
+    val mergeInsert = Value
```

Comment:
This is part about merge is not used yet, it's just easier to commit the test infrastructure in this PR.